modules/layers/src/screen-grid-layer/screen-grid-layer.js

```diff
@@ -261,17 +261,17 @@ export default class ScreenGridLayer extends Layer {
   }
   // Process 'data' and build positions and weights Arrays.
   _processData() {
-    const {data, getPosition} = this.props;
+    const {getPosition} = this.props;
     const positions = [];
     const colorWeights = [];
     const {weights} = this.state;
 
-    for (const point of data) {
-      const position = getPosition(point);
+    this.iterateData((object, context) => {
+      const position = getPosition(object, context);
```

Comment:
Unrelated: Pushing onto arrays can be pretty expensive, wonder if we can get rid of that?